internal/interfaces/grpc/handler/operator.go

```diff
@@ -726,20 +759,28 @@ func (o operatorHandler) listMarket(ctx context.Context, req *pb.ListMarketReque
 				return nil, err
 			}
 
-			pbMarketInfos := make([]*pb.MarketInfo, len(marketInfos), len(marketInfos))
+			pbMarketInfos := make([]*pb.MarketInfo, 0, len(marketInfos))
```

Comment:
could you rever this? the existing way of filling the slice is actually less performant than this one. We should move to this solution everywhere else. I remember louis have already pointed this out previously.

Reply:
I am not sure that this is correct, i did benchmark and results are pretty much similar, and IMHO i think using append is more idiomatic.

````
func BenchmarkAppendOne(b *testing.B) {
	for i := 0; i < b.N; i++ {
		appendOne()
	}
}

func BenchmarkAppendTwo(b *testing.B) {
	for i := 0; i < b.N; i++ {
		appendTwo()
	}
}

func appendOne() {
	markets := make([]domain.Market, 0)
	for i := 0; i < 100; i++ {
		markets = append(markets, domain.Market{
			AccountIndex: 0,
			BaseAsset:    "",
			QuoteAsset:   "",
			Fee:          0,
			Tradable:     false,
			Strategy:     mm.MakingStrategy{},
			Price:        domain.Prices{},
		})
	}

	marketInfos := make([]MarketInfo, 0, len(markets))

	for _, market := range markets {
		marketInfos = append(marketInfos, MarketInfo{
			AccountIndex: uint64(market.AccountIndex),
			Market: Market{
				BaseAsset:  market.BaseAsset,
				QuoteAsset: market.QuoteAsset,
			},
			Fee: Fee{
				BasisPoint: market.Fee,
			},
			Tradable:     market.Tradable,
			StrategyType: market.Strategy.Type,
			Price:        market.Price,
		})
	}
}

func appendTwo() {
	markets := make([]domain.Market, 0)
	for i := 0; i < 100; i++ {
		markets = append(markets, domain.Market{
			AccountIndex: 0,
			BaseAsset:    "",
			QuoteAsset:   "",
			Fee:          0,
			Tradable:     false,
			Strategy:     mm.MakingStrategy{},
			Price:        domain.Prices{},
		})
	}

	marketInfos := make([]MarketInfo, len(markets))

	for index, market := range markets {
		marketInfos[index] = MarketInfo{
			AccountIndex: uint64(market.AccountIndex),
			Market: Market{
				BaseAsset:  market.BaseAsset,
				QuoteAsset: market.QuoteAsset,
			},
			Fee: Fee{
				BasisPoint: market.Fee,
			},
			Tradable:     market.Tradable,
			StrategyType: market.Strategy.Type,
			Price:        market.Price,
		}
	}
}
````

````
///////////////////////
BenchmarkAppendOne
BenchmarkAppendOne-8   	   66777	     20478 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   73107	     16236 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   65511	     16411 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   66888	     16502 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   58797	     17023 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   60243	     18762 ns/op
PASS

////////////////////
BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   54678	     21754 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   59526	     17693 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   65001	     16569 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   63954	     18000 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   62292	     16478 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   69199	     17819 ns/op
PASS
````

Reply:
did you take a look at this https://stackoverflow.com/a/38654841?

Reply:
Discussed offline with @altafan and improved testing, PSB:

```
var markets []domain.Market

func init() {
	markets := make([]domain.Market, 0)
	for i := 0; i < 100000; i++ {
		markets = append(markets, domain.Market{
			AccountIndex: 0,
			BaseAsset:    "",
			QuoteAsset:   "",
			Fee:          0,
			Tradable:     false,
			Strategy:     mm.MakingStrategy{},
			Price:        domain.Prices{},
		})
	}
}

func BenchmarkAppendOne(b *testing.B) {
	for i := 0; i < b.N; i++ {
		appendOne(markets)
	}
}

func BenchmarkAppendTwo(b *testing.B) {
	for i := 0; i < b.N; i++ {
		appendTwo(markets)
	}
}

func appendOne(markets []domain.Market) {

	marketInfos := make([]MarketInfo, 0, len(markets))

	for _, market := range markets {
		marketInfos = append(marketInfos, MarketInfo{
			AccountIndex: uint64(market.AccountIndex),
			Market: Market{
				BaseAsset:  market.BaseAsset,
				QuoteAsset: market.QuoteAsset,
			},
			Fee: Fee{
				BasisPoint: market.Fee,
			},
			Tradable:     market.Tradable,
			StrategyType: market.Strategy.Type,
			Price:        market.Price,
		})
	}
}

func appendTwo(markets []domain.Market) {
	marketInfos := make([]MarketInfo, len(markets), len(markets))

	for index, market := range markets {
		marketInfos[index] = MarketInfo{
			AccountIndex: uint64(market.AccountIndex),
			Market: Market{
				BaseAsset:  market.BaseAsset,
				QuoteAsset: market.QuoteAsset,
			},
			Fee: Fee{
				BasisPoint: market.Fee,
			},
			Tradable:     market.Tradable,
			StrategyType: market.Strategy.Type,
			Price:        market.Price,
		}
	}
}



BenchmarkAppendOne
BenchmarkAppendOne-8   	172570485	         6.78 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	168643347	         6.88 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	174481154	         6.80 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	175577139	         6.79 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	179474529	         6.83 ns/op
PASS

/////////////////////////////////////

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	174470482	         6.75 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	173583262	         6.75 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	172803805	         7.20 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	174627186	         6.71 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	174373942	         6.85 ns/op
PASS
```

@tiero what do you think?